sklearn/metrics/tests/test_pairwise.py

```diff
@@ -192,18 +193,18 @@ def test_pairwise_boolean_distance(metric):
         pairwise_distances(X.astype(bool), Y=Y, metric=metric)
 
     # Check that no warning is raised if X is already boolean and Y is None:
-    with pytest.warns(None) as records:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
```

Comment:
From the commit that introduced this test: https://github.com/scikit-learn/scikit-learn/commit/c7ce6757d35d50c59b9e3d59ece755e95dd0e602

```suggestion
        warnings.simplefilter("error", DataConversionWarning)
```

For reference, it's best to look at `git blame` to figure the commit to figure out which warning we were catching.